### DISCUSSION TITLE
Empty collection is non-empty after calling repository.create()

### DISCUSSION THREAD

**Discussion by mjroeleveld** [mikro-orm/mikro-orm#5419] [Q&A]
This seems very weird. I'm calling `EntityRepository.create(entity)` which causes collections (`pushNotificationTokens` and `deletionHolds`) to be populated with an item, whereas before calling `create()` it it had a size of `0`. 

What can be causing this?

<img width="1073" alt="Screenshot 2024-04-05 at 16 42 28" src="https://github.com/mikro-orm/mikro-orm/assets/5803704/ecfbb189-510f-47dc-82ef-be09002e47de">

**Comment by B4nan** [comment]
The collection state is synced with the identity map automatically, so if you have a 1:m collection and an entity that has corresponding m:1 property, the collection will automatically contain the item (and same works for all kinds of relations).

https://mikro-orm.io/docs/propagation

**Comment by mjroeleveld** [comment]
Thanks. However, there are no corresponding entities. I cleared the entity manager to be sure, the result is the same:

<img width="1165" alt="Screenshot 2024-04-05 at 19 11 55" src="https://github.com/mikro-orm/mikro-orm/assets/5803704/34eb45f9-4a67-4c92-ba45-4250027c4f97">

**Reply by B4nan** [reply]
By clearing entity manager you don't change any state, the changes are propagated eagerly, when you assign to the properties, which probably happens during the `em.create` call.

**Comment by mjroeleveld** [comment]
@B4nan I indeed suspect it has to do with the way I initialize the collection, with a ref. Does that work at all?

```typescript
const model: UserModel = {
      id: copy.id,
      // ...
      pushNotificationTokens: new Collection<UserPushNotificationTokenModel>(
        ref(UserModel, copy.id),
        copy.pushNotificationTokens.tokens.map((t) => ({
          user: ref(UserModel, copy.id),
          token: t.token,
        })),
      ),
      deletionHolds: new Collection<DeletionHoldModel>(
        ref(UserModel, copy.id),
        copy.deletionHolds.getHolds().map((h) => ({
          user: ref(UserModel, copy.id),
          ...h,
        })),
      ),
    };
    return this.orm.em.create(UserModel, model);
```

**Reply by B4nan** [reply]
Why are you doing that? Just pass an array of POJOs in place of a collection. Also you don't need to link the M:1 property, that will be filled automatically. This should work:

```ts
const model: UserModel = {
  id: copy.id,
  // ...
  pushNotificationTokens: copy.pushNotificationTokens.tokens,
  deletionHolds: copy.deletionHolds.getHolds(),
};
return this.orm.em.create(UserModel, model);
```

**Reply by mjroeleveld** [reply]
But that doesn't satisfy the type `Collection<Type>` right?
<img width="403" alt="Screenshot 2024-04-06 at 09 36 14" src="https://github.com/mikro-orm/mikro-orm/assets/5803704/773bca17-5253-4d8d-a03d-b811898130c8">

**Reply by B4nan** [reply]
You are not supposed to be passing a collection instance there. Maybe if you start providing code I can try instead of screenshots, I can actually help you. Right now it feels like you are testing me if I know my own code :D

**Reply by mjroeleveld** [reply]
Sorry I don't have more code to provide than I already sent. If I cannot pass a collection instance; how can I provide the POJO such that it satisfies the typing?

**Reply by B4nan** [reply]
You can pass it, but you don't need to - you are not supposed to be doing that, passing an array of POJOs is the suggested way. I don't think there is a single place in docs that would show passing collection instances. 

Also, I am not sure what your original problem is, do you have an actual problem or is it all just confusion? Because to me your OP sounds like you are just confused by how the identity map works. You always get a consistent entity graph back. If there is a bidirectional relation, it will be valid on both sides - you have a m:1 property, and it will automatically propagate to the 1:m inverse side.

**Reply by mjroeleveld** [reply]
I have separate persistence/domain models so I need to map from/to MikroOrm models to my own classes. 

This works:
```typescript
const model: RequiredEntityData<UserModel> = {
  id: copy.id,
  // ...
  pushNotificationTokens: copy.pushNotificationTokens.tokens,
  deletionHolds: copy.deletionHolds.getHolds(),
};
return this.orm.em.create(UserModel, model);
```
